truffle/src/main/java/org/jruby/truffle/language/literal/NilNode.java

```diff
@@ -32,7 +32,7 @@ public Object execute(VirtualFrame frame) {
 
     @Override
     public Object isDefined(VirtualFrame frame) {
-        return create7BitString("nil", UTF8Encoding.INSTANCE);
+        return stringLiterals().NIL.getString();
```

Comment:
This is the motivation for this work - we want to be able to refer to a compilation final immutable rope which we can package up in a tiny, escape analysable string object very quickly.